src/utilities/byProgram.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2019
+ */
+
+/**
+ * Generic helper methods to handle usage of MasterLists which
+ * are programs and program based requisitions and stock takes.
+ */
+
+/**
+ * Finds if the currently logged in store is a store which uses programs.
+ * If true then: there exists a master list which is a program, which has
+ * a store tag in programSettings which also exists in the tags for this
+ * store.
+ * @param  {Object}  settings
+ * @param  {Realm}   database
+ * @return {boolean} indicator of the currently logged in store using programs
+ */
+const thisStoreUsesPrograms = (settings, database) => {
+  const thisStoresTags = settings.get('ThisStoreTags');
+  const programs = database.objects('MasterList').filtered('isProgram = $0', true);
+  const hasAProgram = programs.find(program => program.canUseProgram(thisStoresTags));
+  return !!hasAProgram;
```

Comment:
canUseProgram should also check that at least one order type exists

Reply:
I will add this, just a tangent: Do you think a there should be an enforcement of an orderType in 4D when creating a store tag? (When you now set a master list to be a program, there is validation that there exists a store tag, do you think extending this to also include requiring an order type is to much?)

Reply:
done b983e71ac72281f8329e9de43dd87dbdf1ce6aac